src/tests/unit/platform/test_platform_process.py

```diff
@@ -86,10 +96,119 @@ def test_process_select(cbcsdk_mock):
+                             POST_PROCESS_SEARCH_JOB_RESP)
+    # mock the GET to check summary search status
+    cbcsdk_mock.mock_request("GET", ("/api/investigate/v2/orgs/test/processes/"
+                                     "summary_jobs/2c292717-80ed-4f0d-845f-779e09470920"),
+                             GET_PROCESS_SUMMARY_RESP)
+    # mock the GET to get summary search results
+    cbcsdk_mock.mock_request("GET", ("/api/investigate/v2/orgs/test/processes/"
+                                     "summary_jobs/2c292717-80ed-4f0d-845f-779e09470920/results"),
+                             GET_PROCESS_SUMMARY_RESP)
+    api = cbcsdk_mock.api
+    guid = 'WNEXFKQ7-0002b226-000015bd-00000000-1d6225bbba74c00'
+    summary = api.select(Process.Summary).where(f"process_guid:{guid}")
+    assert isinstance(summary, SummaryQuery)
+    summary._query_token = 'something'
+    with pytest.raises(ApiError) as ex:
+        summary._submit()
+    assert 'Query already submitted:' in ex.value.message
+    summary._query_token = None
+    with pytest.raises(ApiError) as ex:
+        summary._run_async_query('someother')
+    assert ex.value.message == 'Async query not properly started'
+
+
+def test_summary_still_querying_zero(cbcsdk_mock):
+    """Testing edge cases for _still_querying"""
+    # mock the POST of a summary search (using same Job ID)
+    cbcsdk_mock.mock_request("POST", "/api/investigate/v2/orgs/test/processes/summary_jobs",
+                             POST_PROCESS_SEARCH_JOB_RESP)
+    # mock the GET to check summary search status
+    cbcsdk_mock.mock_request("GET", ("/api/investigate/v2/orgs/test/processes/"
+                                     "summary_jobs/2c292717-80ed-4f0d-845f-779e09470920"),
+                             GET_PROCESS_SUMMARY_RESP_ZERO_CONTACTED)
+    api = cbcsdk_mock.api
+    guid = 'WNEXFKQ7-0002b226-000015bd-00000000-1d6225bbba74c00'
+    summary = api.select(Process.Summary).where(f"process_guid:{guid}")
+    assert summary._still_querying() is True
+
+
+def test_summary_still_querying(cbcsdk_mock):
+    """Testing edge cases for _still_querying"""
+    # mock the POST of a summary search (using same Job ID)
+    cbcsdk_mock.mock_request("POST", "/api/investigate/v2/orgs/test/processes/summary_jobs",
+                             POST_PROCESS_SEARCH_JOB_RESP)
+    # mock the GET to check summary search status
+    cbcsdk_mock.mock_request("GET", ("/api/investigate/v2/orgs/test/processes/"
+                                     "summary_jobs/2c292717-80ed-4f0d-845f-779e09470920"),
+                             GET_PROCESS_SUMMARY_RESP_STILL_QUERYING)
     api = cbcsdk_mock.api
     guid = 'WNEXFKQ7-0002b226-000015bd-00000000-1d6225bbba74c00'
     summary = api.select(Process.Summary).where(f"process_guid:{guid}")
+    assert summary._still_querying() is True
+    summary._search()
+    summary._timed_out = True
+    summary._search()
```

Comment:
This needs an assert or some kind of verification that having set the _timed_out the following search function call did something different or completed

Reply:
Actually I tried to cover one part of the _search() function, but this did not do the trick for some reason, so removing the last two lines - there are not doing anything.

Reply:
@emitreva-cb There is still no validation after `summary._search()`. Removing the two lines is a start however there needs to be some verification of what that line is doing. Either a comment saying this will raise an error if there is a mistake/bug or verify that the search completed successfully